cl/recap/tasks.py

```diff
@@ -2300,6 +2329,89 @@ def fetch_docket_by_pacer_case_id(session, court_id, pacer_case_id, fq):
     }
 
 
+@app.task(
+    bind=True,
+    autoretry_for=(PacerLoginException, RedisConnectionError),
+    max_retries=5,
+    interval_start=5,
+    interval_step=5,
+    ignore_result=True,
+)
+def fetch_appellate_docket(self, fq_pk):
+    """Fetches an appellate docket from PACER using the docket number
+    associated with the provided Fetch Queue record to attempt the purchase.
+
+    :param fq_pk: The PK of the Fetch Queue to update.
+    :return: None
+    """
+    fq = PacerFetchQueue.objects.get(pk=fq_pk)
+    court_id = fq.court_id or getattr(fq.docket, "court_id", None)
+
+    # Check court connectivity, if fails retry the task, hopefully, it'll be
+    # retried in a different not blocked node
+    if not is_pacer_court_accessible(court_id):
+        if self.request.retries == self.max_retries:
+            msg = f"Blocked by court: {court_id}"
+            mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+            self.request.chain = None
+            return None
+        raise self.retry()
+
+    async_to_sync(mark_pq_status)(fq, "", PROCESSING_STATUS.IN_PROGRESS)
+
+    session_data = get_pacer_cookie_from_cache(fq.user_id)
+    if session_data is None:
+        msg = f"Cookie cache expired before task could run for user: {fq.user_id}"
+        mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+        self.request.chain = None
+        return None
+
+    s = ProxyPacerSession(
+        cookies=session_data.cookies, proxy=session_data.proxy_address
+    )
+
+    docket_number = fq.docket_number or getattr(
+        fq.docket, "docket_number", None
+    )
+    start_time = now()
+    try:
+        result = get_appellate_docket_by_docket_number(
+            docket_number=docket_number,
+            court_id=map_cl_to_pacer_id(court_id),
+            session_data=s,
+            **get_fq_appellate_docket_kwargs(fq),
+        )
+    except (requests.RequestException, ReadTimeoutError) as exc:
+        msg = "Network error while purchasing docket for fq: %s."
+        if self.request.retries == self.max_retries:
+            mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+            self.request.chain = None
+            return None
+        mark_fq_status(
+            fq, f"{msg}Retrying.", PROCESSING_STATUS.QUEUED_FOR_RETRY
+        )
+        raise self.retry(exc=exc)
+    except ParsingException:
+        msg = "Unable to purchase docket for fq: %s."
+        mark_fq_status(fq, msg, PROCESSING_STATUS.FAILED)
+        self.request.chain = None
+        return None
+
```

Comment:
I think it would be a good idea to also catch `PacerLoginException` here since it's a potential error that can occur while retrieving the docket. This way, the `FQ` is marked as failed with a related message in case it runs out of retries.

Reply:
Great catch. I'll add the logic to handle PacerLoginException errors. Thanks